### ISSUE TITLE
Remove auto-generated/injected anchorjs (/assets/javascript/anchor-js)

### ISSUE THREAD

**Issue by laukstein** [github/pages-gem#482]
Issue discussed in https://github.com/jekyll/jekyll/issues/6358

When used [pages-gem recommended](https://github.com/github/pages-gem#usage) **Gemfile**

    gem "github-pages", group: :jekyll_plugins

even when this **Gemfile** and minimal **_config.yml** are the only files in Jekyll dir, on `jekyll serve` it will generate **_site/assets** with [anchorjs](https://github.com/bryanbraun/anchorjs). Why?

![grabilla g13256](https://user-images.githubusercontent.com/183966/30325522-bfeeb4a8-97cd-11e7-8146-ef7c24095a57.png)

The only way I found to remove **_site/assets** was to remove **Gemfile** or have it `gem "github-pages"` without ~`group: :jekyll_plugins`~.

**Comment by benbalter** [comment]
@laukstein I believe if you set `theme: none` or `theme: ""` it should not use the Primer theme (which is the source of the asset).

**Comment by laukstein** [comment]
@benbalter, Jekyll fails when used `theme: none` or `theme: ""` in **Gemfile**, see

with `theme: none`

> Configuration file: .../_config.yml
> jekyll 3.5.2 | Error:  The none theme could not be found.

with `theme: ""`

> Configuration file: ../_config.yml
> jekyll 3.5.2 | Error:  The  theme could not be found.

**Comment by laukstein** [comment]
@mattr, it gets me generated without `theme` key. See the above comment, the issue might be related to `group: :jekyll_plugins`.

**Comment by mattr-** [comment]
I don't put the `github-pages` gem in the `jekyll-plugins` group either, so maybe that's the ultimate fix? No idea.

**Comment by laukstein** [comment]
@mattr-, [github/pages-gem](https://github.com/github/pages-gem) recommendation is to use `group: :jekyll_plugins`, so #482 is a issue.

**Comment by benbalter** [comment]
@mattr- https://github.com/pages-themes/primer is the default theme, set by the GitHub Pages Gem when no other theme is specified, so that workflows can be as simple as committing markdown files and having a site. It is largely independent, unless of course, you care about extraneous assets i the built site.

@laukstein What if you add the following to your config (and remove the theme suggestion above):

```yml
exclude:
  - assets/javascript/anchor-js
```

I believe that should exclude the extraneous assets from appearing in your built site.

**Comment by laukstein** [comment]
@benbalter,

>     exclude:
>         - assets/javascript/anchor-js

it will still generate **_site/assets/javascript/anchor-js**.

**Comment by benbalter** [comment]
@parkr ☝️ is it possible themes don't respect the `excludes` directive?

**Comment by DirtyF** [comment]
@benbalter According to my first tests, it seems so.

**Comment by mattr-** [comment]
> https://github.com/pages-themes/primer is the default theme, set by the GitHub Pages Gem when no other theme is specified, so that workflows can be as simple as committing markdown files and having a site.

@benbalter 😕  I'm not putting `github-pages` in the `jekyll-plugins` group of the Gemfile in my site. Is this what keeps these files from being generated since the hook doesn't get run? I don't have a `theme` key in my config.  I'd like to make sure I've got it run so I don't give folks wrong info in the future. 🙂

**Comment by benbalter** [comment]
> Is this what keeps these files from being generated since the hook doesn't get run?

Correct. When in the `jekyll_plugins` group it's required earlier in the boot process, so https://github.com/github/pages-gem/blob/master/lib/github-pages.rb#L18-L20 can fire and set the `theme` config directive, before Jekyll configures the theme in `site`. The `jekyll_plugins` group allows the plugin to set configuration defaults and overrides prior to the site object being setup, and before other plugins.

**Comment by mattr-** [comment]
TIL. Thanks! :heart:

**Comment by parkr** [comment]
> is it possible themes don't respect the excludes directive

Definitely possible! Feel free to close this and opens bug report to solve that in Jekyll.

We have no way of knowing whether someone has used this resource so the best thing is probably to re-evaluate having the resource in the theme, consolidating the resource into another resource so it's less obvious it's not useful (heh), or ensuring it is excludable.

**Comment by laukstein** [comment]
If closing this task, please referrer here the actual (new) issue.

**Comment by jmlemetayer** [comment]
A simple _ugly_ fix is to delete the [line 19](https://github.com/github/pages-gem/blob/a4626834d00beba7f1d87357b95b7c25648bb82f/lib/github-pages/configuration.rb#L19) of the `lib/github-pages/configuration.rb`:

```ruby
      "theme"    => "jekyll-theme-primer",
```

Here is the new generated tree:

```
.
├── Gemfile
├── Gemfile.lock
├── index.html
└── _site
    └── index.html
```

Instead of:

```
.
├── Gemfile
├── Gemfile.lock
├── index.html
└── _site
    ├── assets
    │   ├── css
    │   │   └── style.css
    │   └── javascript
    │       └── anchor-js
    │           ├── anchor.js
    │           ├── anchor.min.js
    │           ├── banner.js
    │           └── docs
    │               ├── anchor.js
    │               ├── favicon.ico
    │               ├── fonts
    │               │   ├── anchorjs-extras.eot
    │               │   ├── anchorjs-extras.svg
    │               │   ├── anchorjs-extras.ttf
    │               │   ├── anchorjs-extras.woff
    │               │   └── fonts.css
    │               ├── grunticon
    │               │   ├── grunticon.loader.js
    │               │   ├── icons.data.png.css
    │               │   ├── icons.data.svg.css
    │               │   ├── icons.fallback.css
    │               │   └── png
    │               │       └── grunticon-link.png
    │               ├── img
    │               │   ├── anchoring-links.png
    │               │   ├── anchorjs_logo.png
    │               │   ├── anchorlinks2.png
    │               │   ├── gh_link.svg
    │               │   ├── gh-link.svg
    │               │   ├── hyperlink.svg
    │               │   ├── link.svg
    │               │   ├── mini-logo.png
    │               │   └── primer-md.png
    │               ├── scripts.js
    │               └── styles.css
    └── index.html
```

I am not a "ruby man" but I think it could be possible to do something like:

```
IF theme is unset
THEN
    theme = jekyll-theme-primer
ELSE IF theme == false OR theme == "none"
THEN
    UNSET theme
FI
```

---

You can check my [docker-github-pages patch](https://github.com/jmlemetayer/docker-github-pages/commit/0082df5b327c46799555c1570ccd157659790173#diff-3254677a7917c6c01f55212f86c57fbf).

**Comment by oscarotero** [comment]
In `_config.yml` this works fine to me:

```yaml
theme: false
```

**Comment by laukstein** [comment]
@oscarotero, I just checked it and in local machine and it gets rid `anchor-js`, so far fine. Unfortunately it would return also error/warning:

> Theme: value of 'theme' in config should be String to use gem-based themes, but got FalseClass

I didn't try, but it might be that GitHub Pages build would fail/break on it.

**Comment by oscarotero** [comment]
@laukstein I have that error too, but github build doesn't break.
This is the project: https://github.com/oscarotero/oscarotero.github.io

**Comment by stevetweeddale** [comment]
If the error irks you, the workaround I'm using for now is setting my theme to be a remote one that I just found with basically nothing in it. So in `_config.yml`:

```
remote_theme: garrettboatman/Blank-Theme-Jekyll
```

**Comment by laukstein** [comment]
@benbalter this issue still must stay open till Github Pages adopts this fix. Doesn't Github Pages depends on landed packages? And if so then this fix still may take time.
